client/my-sites/hosting/staging-site-card/index.js

```diff
@@ -85,7 +84,8 @@ export const StagingSiteCard = ( { currentUserId, disabled, siteId, siteOwnerId,
 	const stagingSite = useMemo( () => {
 		return stagingSites && stagingSites.length ? stagingSites[ 0 ] : [];
 	}, [ stagingSites ] );
-	const hasSiteAccess = useHasSiteAccess( stagingSite.id );
+	const hasSiteAccess =
+		! stagingSite.id || Boolean( stagingSite.id && stagingSite?.user_has_permission );
```

Comment:
I think we could remove `stagingSite.id` from the second part of the condition.
```suggestion
		! stagingSite.id || Boolean( stagingSite?.user_has_permission );
```